aiohttp/helpers.py

```diff
@@ -1038,3 +1041,24 @@ def parse_http_date(date_str: Optional[str]) -> Optional[datetime.datetime]:
             with suppress(ValueError):
                 return datetime.datetime(*timetuple[:6], tzinfo=datetime.timezone.utc)
     return None
+
+
+def make_an_interface_bound_socket(
+    network_interface: str,
+    family: int = socket.AF_INET,
+    local_addr: Optional[Union[Tuple[Any, ...], str]] = None,
+) -> socket.socket:
+    """Return a socket, bounded to a network interface.
+
+    Addresses can be either tuples of varying lengths
+    (AF_INET, AF_INET6, AF_NETLINK, AF_TIPC)
+    or strings (AF_UNIX).
+    """
+    sock = socket.socket(family, socket.SOCK_STREAM)
```

Comment:
It probably makes sense to add
```suggestion
    assert SOCKET_SUPPORTS_BINDING_TO_DEVICE
    sock = socket.socket(family, socket.SOCK_STREAM)
```